PhysicsTools/Heppy/python/analyzers/objects/METAnalyzer.py

```diff
@@ -85,23 +113,70 @@ def makeTkMETs(self, event):
                 if dochs and  pvflag>0:
                     chargedchs.append(pxy)
 
-                if doloose and pvflag>1:
+                ##CHS=loose
+                if doloose and pvflag>=1:
                     chargedPVLoose.append(pxy)
 
-                if dotight and pvflag>2:
+                if dotight and pvflag>=2:
                     chargedPVTight.append(pxy)
 
+                if doNoPV and pvflag >= 0:
+                    chargedNoPV.append(pxy)
+
+                if doPVUsedInFit and pvflag== 3:
+                    chargedPVUsedInFit.append(pxy)
+
+                #adding pvAssociationQuality stuff
+                pvAssflag = pfcand.pvAssociationQuality()
+
+                if doUsedInFitTight and pvAssflag == 7 :
+                    chargedUsedInFitTight.append(pxy)
+
+                if doUsedInFitLoose and pvAssflag == 6 :
```

Comment:
wouldn't it make more sense to do it with `>=` instead of `==` ?
also, shouldn't you also check `pfRef.key()` ? the association quality only tells you how well it's associated to its PV, it doesn't tell you whether it's PV[0] or another one.